src/Microsoft.ServiceFabric.Actors/Runtime/KVStoRCMigrationActorStateProvider.cs

```diff
@@ -30,8 +31,12 @@ public class KVStoRCMigrationActorStateProvider :
         private static readonly DataContractSerializer LogicalTimestampDataContractSerializer = new DataContractSerializer(typeof(LogicalTimestamp));
 
         private string traceId;
-        private IStatefulServicePartition servicePartition;
         private ReliableCollectionsActorStateProvider rcStateProvider;
+        private IStatefulServicePartition servicePartition;
+        private IReliableDictionary2<string, byte[]> metadataDictionary;
+        private bool isMetadataDictInitialized = false;
```

Comment:
We check that the state provider initialization is complete at the start of migrationOrchestrator.StartMigration. Since SaveStateAsync() is always called after that, we do not need to check this again.